trivup/apps/SchemaRegistryApp.py

```diff
@@ -28,96 +28,115 @@
 
 from trivup import trivup
 from trivup.apps.KafkaBrokerApp import KafkaBrokerApp
+from trivup.apps.SslApp import SslApp
 
-import uuid
 import requests
-import subprocess
 
 
-class SchemaRegistryApp (trivup.App):
+sr_image = 'confluentinc/cp-schema-registry'
+
+
+class SchemaRegistryApp (trivup.DockerApp):
     """ Confluent Schema Registry app.
         Depends on KafkaBrokerApp.
         Requires docker. """
 
-    default_image = 'confluentinc/cp-schema-registry:5.2.1'
+    default_version = '5.3.0'
+
+    default_app_conf = {
+        'debug': True,
+        'host.name': 'localhost',
+        'listeners': 'http://0.0.0.0:8081'
+    }
 
-    def __init__(self, cluster, conf=None, on=None):
+    @staticmethod
+    def _conf_to_env(app_conf):
+        return ["SCHEMA_REGISTRY_{}={}"
+                .format(key.replace(".", "_").upper(), value)
+                for key, value in app_conf.items()]
+
+    def __init__(self, cluster, conf=None):
         """
         @param cluster     Current cluster
         @param conf        Configuration dict, see below.
         @param on          Node name to run on
 
         Supported conf keys:
            * version - Confluent Platform version to use.
+           * enable_auth - Enable Http basic auth
            * port_base - Low TCP port base to start allocating from (random)
-           * image - docker image to use
-           * conf - schema-registry docker image config strings (NOT USED)
+           * conf - Dictionary of Confluent Schema Registry configs
         """
-        super(SchemaRegistryApp, self).__init__(cluster, conf=conf, on=on)
+        print(conf.get('enable_auth'))
+        # Handle all docker specific after setting the context
+        super(SchemaRegistryApp, self).__init__(cluster,
+                                                sr_image,
+                                                conf.get('version',
+                                                         self.default_version),
+                                                conf)
+
+        protocol = 'http'
+        # Find available port on the host and expose it
+        port = trivup.TcpPortAllocator(cluster).next(
+            self, self.conf.get('port_base', None))
+        self.expose_port(self.docker_args, 8081, port)
 
-        if self.conf.get('image', '') == '':
-            self.conf['image'] = self.default_image
+        app_conf = dict(self.default_app_conf, **conf.get('conf', {}))
 
-        self.conf['container_id'] = 'trivup_sr_%s' % str(uuid.uuid4())[0:7]
         kafka = cluster.find_app(KafkaBrokerApp)
-        if kafka is None:
+        if kafka:
```

Comment:
You should avoid if blocks (extra indentation) for the sunny path.
So instead of:
```
if some_err:
   bail out
else:
   the normal case
```

(the else block is unnecessary) 

do what the original code did:
```
if some_err:
   bail out

the normal case
```